### PR TITLE
Added unit-of-work

### PR DESCRIPTION
**Pull request by MartinSchmidt**
Added the following and refactored code:
- GrpcTests that runs integrations test on the gRPC service.
- Unit-of-work and repositories including integration tests with database

### CODE REVIEW

**Review by duizer — CHANGES_REQUESTED** [review]
The github workflow does not finish within 2hours.

**Review by duizer — CHANGES_REQUESTED** [review]
The github workflow does not finish within 2hours.

**Review by tnickelsen — APPROVED** [review]

File: src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/PostgresDatabaseFixture.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private PostgreSqlContainer _postgreSqlContainer;
+
+    public PostgresDatabaseFixture()
+    {
+        _postgreSqlContainer = new PostgreSqlBuilder()
+            .WithImage("postgres:15")
+            .Build();
+    }
+
+    public virtual Task InitializeAsync()
```
Comment (duizer) [inline src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/PostgresDatabaseFixture.cs:20]: Why is this marked as virtual?

File: src/ProjectOrigin.Wallet.Server/Program.cs [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
 var app = builder.Build();
 
+startup.Configure(app, builder.Environment);
+
 DatabaseUpgrader.Upgrade(app.Configuration.GetConnectionString("Database"));
 
 // Configure the HTTP request pipeline.
-app.MapGrpcService<ExternalWalletService>();
+app.MapGrpcService<WalletService>();
 app.MapGet("/", () => "Communication with gRPC endpoints must be made through a gRPC client. To learn how to create a client, visit: https://go.microsoft.com/fwlink/?linkid=2086909");
```
Comment (duizer) [inline src/ProjectOrigin.Wallet.Server/Program.cs:21]: This line is also defined in Startup

File: src/ProjectOrigin.Wallet.Server/Program.cs [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
 
 var app = builder.Build();
 
+startup.Configure(app, builder.Environment);
+
 DatabaseUpgrader.Upgrade(app.Configuration.GetConnectionString("Database"));
 
 // Configure the HTTP request pipeline.
-app.MapGrpcService<ExternalWalletService>();
+app.MapGrpcService<WalletService>();
```
Comment (duizer) [inline src/ProjectOrigin.Wallet.Server/Program.cs:20]: Should this be in Startup?

File: src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/GrpcHelpers/ForwardLoggingProvider.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+//
+// https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/Helpers/ForwardingLoggerProvider.cs
+#endregion
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace ProjectOrigin.Wallet.IntegrationTests.TestClassFixtures.GrpcHelpers
+{
+    internal class ForwardingLoggerProvider : ILoggerProvider
```
[the remark this answers is not in the document]
Comment (MartinSchmidt) [reply src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/GrpcHelpers/ForwardLoggingProvider.cs:24]: It ensures that logs from the running service can be forwarded to the log output when running the tests.

File: src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/GrpcTestFixture.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            _channel?.Dispose();
+            _handler?.Dispose();
+            _host?.Dispose();
+            _server?.Dispose();
+        }
+
+        public IDisposable GetTestContext(ITestOutputHelper outputHelper)
+        {
+            return new GrpcTestContext<TStartup>(this, outputHelper);
+        }
```
Comment (duizer) [inline src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/GrpcTestFixture.cs:127]: Is the GrpcTestContext needed? This method is never called and it is the only place where the GrpcTestContext is constructed.

**Review by duizer — APPROVED** [review]

**Pull request merged**